task_sdk/src/airflow/sdk/execution_time/comms.py

```diff
@@ -117,7 +117,7 @@ def from_xcom_response(cls, xcom_response: XComResponse) -> XComResult:
         XComResponse is autogenerated from the API schema, so we need to convert it to XComResult
         for communication between the Supervisor and the task process.
         """
-        return cls(**xcom_response.model_dump())
+        return cls(**xcom_response.model_dump(), type="XComResult")
```

Comment:
type has a default -- we shouldn't need to make this change.

Reply:
If this was just to make an idea happy then try this on L110:

```python
    kind: Literal["XComResult"] = Field(init=False, default="XComResult")
```

Reply:
I will remove it, we do not need it